samples/v1/types_config.go

```diff
@@ -64,6 +64,20 @@ type ConfigSpec struct {
 	// content but the operator will not recreate(or update) anything
 	// listed here.
 	SkippedTemplates []string `json:"skippedTemplates,omitempty" protobuf:"bytes,6,opt,name=skippedTemplates"`
+
+	// skippedhelmcharts specifies names of helm charts that should NOT be
+	// managed. Admins can use this to allow them to delete content
+	// they don’t want. They will still have to MANUALLY DELETE the
+	// content but the operator will not recreate(or update) anything
+	// listed here. Few examples of the name of helmcharts which can be skipped are
+	// 'redhat-redhat-perl-imagestreams','redhat-redhat-nodejs-imagestreams','redhat-nginx-imagestreams',
```

Comment:
Are the names always lowercase and hyphen separated? Can any other characters be used for the names? I assume these are all valid Kubernetes resource names?

Reply:
These names won't change, and it wont be always lower case or hyphen separated, but however the imagestream owner make it or name it while making the helmchart, so cant put that check here. and yes these are valid kubernetes resource names

Reply:
So we can check for a valid kube resource name? We can use a CEL expression to explain that. And one to check the maximum length too

```
// +kubebuilder:validation:XValidation:rule="self.all(x, x.matches('^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$'))",message="skipperHelmCharts must be a valid Kubernetes resource name. May contain only lowercase alphanumeric characters, hyphens and periods, and each period separated segment must begin and end with an alphanumeric character."
// +kubebuilder:validation:XValidation:rule="self.all(x, x.size() > 0 && x.size() < 254)",message="skippedHelmCharts must be non-empty and at most 253 characters in length"
```

Reply:
You should also include the explanation of the format in the godoc

Reply:
okay updated it